### PR TITLE
grpc-Support XpringKit Swift SDK

### PR DESCRIPTION
**Pull request by dangell7**
Long term, xpring-common-protocol-buffers is going to disappear and be replaced with an official and well reviewed implementation by the rippled team.

The implementation in rippled is being done here: ripple/rippled#3159.

To access the new rippled protocol buffers, a new submodule is added. Note that the submodule is cjcobb23/rippled (rather than ripple/rippled) since I can't figure out how to get a submodule to track an open PR. I filed an Asana task to track this update here: https://app.asana.com/0/1143722888069470/1155647690941937.

This change segregates the protocol buffers so that we can do the transition in several phases. As of now:

/generated/.proto : 'new' protocol buffers from rippled. Unused, as of now.
/generated/legacy/.proto : The same protocol buffers from xpring-common-protocol-buffers which we have been using.
When we have fully migrated the library to rippled's protocol buffers we can simply delete the legacy folder to delete the migration.

### CODE REVIEW

**Comment by keefertaylor** [comment]
I think that you may have pulled a newer version of `xpring-common-js` that we're ready to use in this repo.  Did you use the `--remote` flag when running `git submodule update`? I usually run:
```
$ git submodule update --init --recursive
```

I think that if you pull `xpring-common-js` at 1710097e899ead9ffdb170b7a37e25bad931cebc this should work. You can do that by:
```
$ cd xpring-common-js
$ git checkout 1710097e899ead9ffdb170b7a37e25bad931cebc
$ cd ..
$ git add -A && git commit -m 'pin xpring-common-js to 1710097e899ead9ffdb170b7a37e25bad931cebc' 
```

File: scripts/clean_project.sh [hunk 1]
```diff
@@ -6,6 +6,7 @@ echo "Cleaning XpringKit Project"
 
 rm -rf *.xcodeproj
 rm -rf XpringKit/generated
+rm -rf XpringKit/generated/Legacy
```
Comment (keefertaylor) [inline scripts/clean_project.sh]: This line is a no op, the `-r` in line 8 is a recursive remove so this directory will also be removed.

**Comment by keefertaylor** [comment]
Apologies, it looks like my review approval didn't actually send my one comment. If you fix that script, I'm happy to merge this.

File: scripts/clean_project.sh (hunk 1 above)
Comment (dangell7) [reply scripts/clean_project.sh]: Oh now I get it. lol

**Comment by keefertaylor** [comment]
Can you update the submodule:
```
cd rippled
git checkout grpcSupport
git pull
cd ..
git add -A
git commit -m 'update rippled commit'
```

At some point soon, rippled should merge this code and then this won't be so brittle.

**Comment by dangell7** [comment]
@keefertaylor no worries. I'm enjoying this now.

**Comment by dangell7** [comment]
Plus, you're an excellent help!! I apologize for all my rude comments before.

**Comment by dangell7** [comment]
@keefertaylor I think the `xpring-common-js` is off now too. Can you help me pin the correct library?

**Comment by keefertaylor** [comment]
Glad you're having a good time! No worries :)

This looks great. Thank you for your work here!

**Comment by keefertaylor** [comment]
> @keefertaylor I think the xpring-common-js is off now too. Can you help me pin the correct library?
I think all is well, since CI is passing.

**Pull request merged**